plutus-ledger-api/test-onchain-evaluation/Main.hs

```diff
@@ -0,0 +1,64 @@
+{-# LANGUAGE LambdaCase        #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TupleSections     #-}
+{-# LANGUAGE TypeApplications  #-}
+
+module Main (main) where
+
+import PlutusCore.Pretty
+import PlutusLedgerApi.Common
+import PlutusLedgerApi.Test.EvaluationEvent
+import PlutusLedgerApi.V1 qualified as V1
+import PlutusLedgerApi.V2 qualified as V2
+
+import Codec.Serialise qualified as CBOR
+import Control.Exception (evaluate)
+import Data.Foldable (toList)
+import System.Directory.Extra (listFiles)
+import System.Environment (getEnv)
+import System.FilePath (isExtensionOf)
+import Test.Tasty
+import Test.Tasty.HUnit
+
+-- | Test cases from a single event dump file
+testOneFile :: FilePath -> IO TestTree
+testOneFile eventFile = do
+    events <- CBOR.readFileDeserialise @ScriptEvaluationEvents eventFile
+    ctxV1 <-
+        either (assertFailure . display) pure $
```

Comment:
I'm not sure this `assertFailure` will do what you want since it occurs outside a `testCase`. I think `assertFailure` just throws an exception which is caught in `testCase` so it can be reported as a failure. 

I think the most idiomatic way to do this might be to use https://hackage.haskell.org/package/tasty-1.4.2.3/docs/Test-Tasty.html#v:withResource

Reply:
Good point.